node/syncblock.go

```diff
@@ -368,6 +400,21 @@ func (localNode *LocalNode) initSyncing() {
 // StartSyncing starts block syncing from current local ledger until it gets to
 // block height stopHeight with block hash stopHash from given neighbors
 func (localNode *LocalNode) StartSyncing(syncStopHash common.Uint256, syncStopHeight uint32, neighbors []*RemoteNode) (bool, error) {
+	cs := chain.DefaultLedger.Store.(*store.ChainStore)
+	fastSyncHeight := syncStopHeight - config.FastSyncPivot
+	log.Info("fast sync begin, stopHeight:", fastSyncHeight)
+	if config.Parameters.SyncMode == "fast" && cs.GetSyncRootHeight() == 0 {
```

Comment:
If a node restarts during fast sync, it cannot resume fast sync right?

Reply:
No

Reply:
We can use the following logic if `config.Parameters.SyncMode == "fast"`:
if `cs.GetSyncRootHeight() == 0`, then use computed `fastSyncHeight`; otherwise use `cs.GetSyncRootHeight()` as stop height. This way it can resume after restarts

Reply:
But it could be dangerous that if user restart node after long time, old root maybe got pruned already.

Reply:
In this case it's user's responsibility to remove chaindb